### PR TITLE
Multiple dock support (for air units, part 2/3)

### PR DESCRIPTION
**Pull request by forcecore**
* Much saner ReturnToBase sequence.

### CODE REVIEW

**Comment by reaperrr** [comment]
Depends on #13732.

**Comment by forcecore** [comment]
Solves https://github.com/OpenRA/OpenRA/issues/5434

File: OpenRA.Mods.Common/Activities/Air/ResupplyAircraft.cs [hunk 1]
```diff
@@ -18,37 +18,35 @@ namespace OpenRA.Mods.Common.Activities
...
-					aircraft.GetResupplyActivities(host)
-					.Append(new AllowYieldingReservation(self))
-					.Append(new TakeOff(self)).Append(NextInQueue).ToArray());
-			}
+			resupplyActivities = aircraft.GetResupplyActivities(host).ToArray();
 		}
 
 		public override Activity Tick(Actor self)
 		{
+			int cnt = 0;
```
Comment (forcecore) [inline OpenRA.Mods.Common/Activities/Air/ResupplyAircraft.cs]: Argh, this got leaked in from my other commit that lets you reload and repair at the same time. If you don't like it I'll revert it to upstream behavior.

Comment (pchote) [inline OpenRA.Mods.Common/Activities/Air/ResupplyAircraft.cs]: To simplify reviewing and git history it would be best if this was left to a later PR.

**Comment by penev92** [comment]
Closing because the dependency's future is unclear and this will likely sit open for ages otherwise.
We can reopen once the dependency is done.

**Pull request closed without merging**